lib/spack/spack/cmd/testsuite.py

```diff
@@ -0,0 +1,221 @@
+        for test in testList:
+            if exclusion in str(test):
+                teststoRemove.append(test)
+    for test in teststoRemove: #remove tests from main test list
+        testList.remove(test)
+    return testList
+
+def testsuite(parser, args):
+    if not args.yamlFile:
+        tty.die("spack testsuite requires a yaml file as argument.")
+    if args.complete:
+        args.log_format='cdash-complete' #use cdash-complete if you want configure, build and test output.
+        cdash = '--log-format=cdash-complete'
+    else:
+        args.log_format='cdash-simple' #use cdash-complete if you want configure, build and test output.
+        cdash = '--log-format=cdash-simple'
+    print args.complete
+    enabledTests = []
+    compilers = []
+    packages = []
+    tests = []
+    testPackages = []
+    versions = []
+    packageVersion = []
+    compilerVersion = []
+    path = ""
+    dashboards = []
+    #designed to use a single file and modify the enabled tests, thus requiring a single file modification.
+    for files in args.yamlFile: #read yaml files which contains description of tests
+        with open(files, 'r') as stream:
+            try:
+                yamlFile= yaml.load(stream)
+            except yaml.YAMLError as exc:
+                print(exc)
+
+            enabledTests= yamlFile['enable']
+            packages= yamlFile['packages'] # list of packages to test. Current libelf, libdwarf, bzip2
+            compilers = yamlFile['compilers']
+            
+            try:
+                path=yamlFile['path']
+            except KeyError as e:
+                tty.msg("Path not found, will use " + str(spack.prefix)+"/var/spack/cdash/")
+
+            try:
+                dashboards = yamlFile['dashboard']
+            except KeyError as e:
+                if path is "":
+                    tty.msg("dashboard url was not found. xml files will be stored at " + str(spack.prefix)+"/var/spack/cdash/")
+                else:
+                    tty.msg("dashboard url was not found. xml files will be stored at path provided.")
+    #creating a list of packages 
+    for package in packages:
+        for pkg in package:
+            if pkg in enabledTests:
+                versions = package[pkg][0]['versions']
+                for version in versions:
+                    # producing packages at available versions. Sample file contains only checksum'd
+                    packageVersion.append(str(pkg)+"@"+str(version))
+
+    #creating a list of compilers
+    for compiler in compilers:
+        for comp in compiler:
+            versions = compiler[comp][0]['versions']
+            for version in versions:
+                # producing compilers at available versions. Sample file contains only checksum'd
+                compilerVersion.append(str(comp)+"@"+str(version))
+    #reducing compiler list to whats actually available on the system
+    compilerVersion = reduceCompilerList(compilerVersion)
```

Comment:
I am not sure I can force a not run. I am sure I can produce a file showing the "not run" but if the compiler is not available It wont install which would produce the build cdash file